hseling-web-cat-and-kittens/hseling_web_cat_and_kittens/main.py

```diff
@@ -35,19 +78,434 @@ def get_server_endpoint():
+
+    elif request.method == 'GET':
+        profileUpdateform.fullname.data = current_user.fullname
+        profileUpdateform.username.data = current_user.username
+        profileUpdateform.email.data = current_user.email
+
+    proImage = url_for('static', filename='media/profile_picture/'+ current_user.profileImage)
+    return render_template('user/profile.html', title='profile', proImage=proImage, form=profileUpdateform)
+
+
+@app.route('/web/upload', methods=['GET','POST'])
+@login_required
+def upload():
+    if request.method == 'POST':
+        authorUser = request.form['user']
+        contentUser = request.form['content']
+        commentUser = request.form['comment']
+        newContentUpload = userUploadForm(author=authorUser, content=contentUser, comment=commentUser)
+        squlitedb.session.add(newContentUpload)
+        squlitedb.session.commit()
+        flash('Your content is updated!')
+        return redirect(url_for('upload'))
+    
+    else:
+        return render_template('user/upload.html')
+
+
+@app.route('/web/edit-content/<int:id>', methods=['GET','POST'])
+@login_required
+def edit(id):
+    editContent = userUploadForm.query.get_or_404(id)
+    if request.method == 'POST':
+        authorEdit = request.form['user']
+        contentEdit = request.form['content']
+        commentEdit = request.form['comment']
+        newEdit = userUploadForm(author=authorEdit, content=contentEdit, comment=commentEdit)
+        squlitedb.session.add(newEdit)
+        squlitedb.session.commit()
+        flash('Your content is updated!')
+        return redirect(url_for('upload'))
+    return render_template('user/edit.html', edits=editContent)
+
+@app.route('/web/view-content/<int:id>')
+@login_required
+def view(id):
+    viewContent = userUploadForm.query.get_or_404(id)
+    return render_template('user/view.html', views=viewContent)
+
+@app.route('/web/history', methods=['GET','POST'])
+@login_required
+def history():
+    allHistory = userUploadForm.query.order_by(userUploadForm.id).all()
+    return render_template('user/history.html', histories=allHistory)
+
+@app.route('/web/')
+def index():
+    return render_template('index.html', title='Home')
+
+@app.route('/web/search', methods=['GET', 'POST'])
+def search():
+    
+    if request.method == "GET":
+        tk = secrets.token_urlsafe()
+        session['csrftoken'] = str(tk)
+        session_csrftoken = session['csrftoken']
+        return render_template('search.html', title='Search', random_token=session_csrftoken)
+
+    else:
+        details = request.form
+        search_token = details['search']
+        csrftoken = details['csrfmiddlewaretoken']
+        print('1', session['csrftoken'])
+        print('2', csrftoken)
+        if csrftoken == session.get('csrftoken', None):
+            frequency = 'freq_all'
+
+            cur = mysql.connection.cursor()
+            cur.execute(f'''SELECT unigrams.{frequency} as frequency, lemmas.lemma as lemma
```

Comment:
Please refrain from introducing SQL injection vulnerabilities. Instead of using f-string in SQL expressions (at least, for user input), use after-comma substitutions:

```python
select_stmt = "SELECT * FROM employees WHERE emp_no = %(emp_no)s"
cursor.execute(select_stmt, { 'emp_no': 2 })
```

Reference: https://dev.mysql.com/doc/connector-python/en/connector-python-api-mysqlcursor-execute.html